DB/src/main/java/io/deephaven/stream/StreamConsumer.java

```diff
@@ -0,0 +1,28 @@
+package io.deephaven.stream;
+
+import io.deephaven.db.v2.sources.chunk.Attributes.Values;
+import io.deephaven.db.v2.sources.chunk.WritableChunk;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Chunk-oriented consumer for streams of data.
+ */
+@FunctionalInterface
+public interface StreamConsumer {
+
+    /**
+     * <p>Accept a batch of rows splayed across per-column {@link WritableChunk chunks} of {@link Values values}.
+     *
+     * <p>Ownership of {@code data} passes to the consumer, which must be sure to
+     * {@link io.deephaven.db.v2.sources.chunk.util.pools.PoolableChunk#close close} each chunk when it's no longer
+     * needed.
+     *
+     * <p>Implementations will generally have a mechanism for determining the expected number and type of input
+     * chunks, but this is not dictated at the interface level.
+     *
+     * @param data Per-column {@link WritableChunk chunks} of {@link Values values}.
+     *             Must all have the same {@link WritableChunk#size() size}.
+     */
+    @SuppressWarnings("unchecked") // There's no actual possibility of heap-pollution, here.
+    void accept(@NotNull WritableChunk<Values>... data);
```

Comment:
No need to be writable.

Reply:
If you want to return it to the pool, they do, unless you want an `instanceof` check for each one. You opposed making `Chunk` implement `PoolableChunk` last time it came up. Also, since they are most certainly writable at the producer, I see no reason not to make it clear that the consumer can do whatever it wants with the chunks it receives.

Reply:
Related, it might be nice to somehow say "these are *not* resettable", since that would interfere with our ability to return them properly.

Reply:
Actually, I take that part (about resettable) back. If we get a resettable chunk, we own the backing storage up to its capacity, and we only need to return the view to the pool. I think that is semantically clean enough for me.